s3/pkg/datastore/yig/config/config.go

```diff
@@ -0,0 +1,155 @@
+package config
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/spf13/viper"
+)
+
+type Config struct {
+	Endpoint   EndpointConfig
+	Log        LogConfig
+	StorageCfg StorageConfig
+	Database   DatabaseConfig
+}
+
+func (config *Config) Parse() error {
+	endpoint := viper.GetStringMap("endpoint")
+	log := viper.GetStringMap("log")
+	storageCfg := viper.GetStringMap("storage")
+	db := viper.GetStringMap("database")
+
+	(&config.Endpoint).Parse(endpoint)
+	(&config.Log).Parse(log)
+	(&config.StorageCfg).Parse(storageCfg)
+	(&config.Database).Parse(db)
+
+	return nil
+}
+
+type CommonConfig struct {
+	Log   LogConfig
+	Cache CacheConfig
+}
+
+func (cc *CommonConfig) Parse() error {
+	log := viper.GetStringMap("log")
+	cache := viper.GetStringMap("cache")
+
+	cc.Log.Parse(log)
+	cc.Cache.Parse(cache)
+
+	return nil
+}
+
+type EndpointConfig struct {
+	Url string
+}
+
+func (ec *EndpointConfig) Parse(vals map[string]interface{}) error {
+	if url, ok := vals["url"]; ok {
+		ec.Url = url.(string)
+		return nil
+	}
+	return errors.New("no url found")
+}
+
+type LogConfig struct {
+	Path  string
+	Level int
+}
+
+func (lc *LogConfig) Parse(vals map[string]interface{}) error {
+	if p, ok := vals["log_path"]; ok {
+		lc.Path = p.(string)
+	}
+	if l, ok := vals["log_level"]; ok {
+		lc.Level = int(l.(int64))
+	}
+	return nil
+}
+
+type StorageConfig struct {
+	CephPath string
+}
+
+func (sc *StorageConfig) Parse(vals map[string]interface{}) error {
+	if p, ok := vals["ceph_dir"]; ok {
+		sc.CephPath = p.(string)
+	}
+	return nil
+}
+
+type CacheConfig struct {
+	Mode              int
+	Nodes             []string
+	Master            string
+	Address           string
+	Password          string
+	ConnectionTimeout int
+	ReadTimeout       int
+	WriteTimeout      int
+	KeepAlive         int
+	PoolMaxIdle       int
+	PoolIdleTimeout   int
+}
+
+func (cc *CacheConfig) Parse(vals map[string]interface{}) error {
```

Comment:
imho, this is too much of code bloat just to hold some configuration variables...unless there is a specific advantage of doing so